test/python/circuit/test_circuit_load_from_qpy.py

```diff
@@ -451,6 +451,50 @@ def test_custom_instruction_with_label(self):
         self.assertEqual(qc.decompose(), new_circ.decompose())
         self.assertEqual([x[0].label for x in qc.data], [x[0].label for x in new_circ.data])
 
+    def test_custom_gate_with_noop_definition(self):
```

Comment:
Maybe it would be good to group these two tests with the `ddt` package or subtests, since they only differ on the type of `empty` and `opaque`. 🙂

Reply:
That's true, but those objects are _so_ core to the test, I'm slightly averse to "hiding" them in the test parameterisation.  Being explicit in a test case isn't necessarily a bad thing, because it's most important that you can see at a glance exactly what the test is testing, and verify that it's obviously correct.  If you think it'd be clearer with them in a `ddt` form I can change it.

(Also, I don't like constructing Terra objects in the collection phase, because if we break them, then the _entire_ test suite fails to collect and nothing runs at all, but that's a minor conern.)

Reply:
It's fine by me if you want to highlight this part of the test 🙂